### PR TITLE
Refactor the offer#update action

### PR DESCRIPTION
**Pull request by erikbjoern**
[PT link - refactor the offer#update action](https://www.pivotaltracker.com/story/show/173417168)
[PT link - Update ReadMe - offer](https://www.pivotaltracker.com/story/show/173413489)

# Changes in this PR:
* The request get its status updated as a direct consequence of an offer being accepted - this is now moved to happen in the Offer model, as an `after_update` method.
 -- makes the controller more skinny and the functionality applies in Rails Console as well
 -- I added missing tests for this functionality
* Readme is corrected regarding offer#show and offer#update responses

### CODE REVIEW

**Comment by erikbjoern** [comment]
Regarding Coveralls - all our code is tested, except Jobs, Mailers and Channels

File: spec/requests/api/offers/update_spec.rb [hunk 1]
```diff
@@ -24,6 +24,11 @@
       it 'responds offer message' do
         expect(response_json['message']).to eq "You accepted help from #{helper.email}"
       end
+
+      it 'sets the status of the request to "active"' do
+        request.reload
+        expect(request.status).to eq "active"
+      end
```
Comment (PaulineBA) [inline spec/requests/api/offers/update_spec.rb:31]: below in that file(nothing you have touched so cant directly comment on the line) one of the it block says 'has a 422 response' but then expect status 500. Doesn't affect anything but I thought we should correct it!

**Review by kermit-klein — APPROVED** [review]

**Review by designerofthing — APPROVED** [review]

**Review by kermit-klein — APPROVED** [review]

File: app/models/offer.rb [hunk 2]
```diff
@@ -15,4 +16,10 @@ def validate_offer
       raise StandardError, 'You cannot offer help on your own request!'
     end
   end
+
+  def update_request_status
```
Comment (jysmys) [inline app/models/offer.rb:20]: for me, i would feel better if the request update is in request model....

**Review by jysmys — APPROVED** [review]
Najs

File: app/models/offer.rb (hunk 2 above)
Comment (erikbjoern) [reply app/models/offer.rb:20]: Agree, did the best of both worlds

**Pull request merged**